custom_components/birthdays/__init__.py

```diff
@@ -45,28 +52,41 @@ async def async_setup(hass, config):
     tasks = [asyncio.create_task(device.update_data()) for device in devices]
     await asyncio.wait(tasks)
 
+    _LOGGER.debug(devices)
+
     return True
 
 
 class BirthdayEntity(Entity):
 
-    def __init__(self, name, date_of_birth, icon, hass):
+    def __init__(self, unique_id, name, date_of_birth, icon, attributes, hass):
         self._name = name
-        self._date_of_birth = date_of_birth
-        self._icon = icon
-        self._age_at_next_birthday = 0
+
+        if unique_id is not None:
+            self._unique_id = slugify(unique_id)
+        else: 
+            self._unique_id = slugify(name)
+
         self._state = None
-        name_in_entity_id = slugify(name)
-        self.entity_id = 'birthday.{}'.format(name_in_entity_id)
+        self._icon = icon
+        self._date_of_birth = date_of_birth
         self.hass = hass
 
+        self._extra_state_attributes = {
+            CONF_DATE_OF_BIRTH: str(self._date_of_birth),
+        }
+
+        if len(attributes) > 0 and attributes is not None:
+            for k,v in attributes.items():
+                self._extra_state_attributes[k] = v
+
     @property
     def name(self):
         return self._name
 
     @property
     def unique_id(self):
-        return '{}.{}'.format(self.entity_id, slugify(self._date_of_birth.strftime("%Y%m%d")))
+        return '{}.{}'.format(self._unique_id, slugify(self._date_of_birth.strftime("%Y%m%d")))
```

Comment:
I tested this on my own system with a few birthdays already in place, but the old values were gone since the new unique_id doesn't default to the same value as the old one (`birthdays.kalle` will be unavailable but there would be a `birthdays.kalle_2` present, and so on). Is there a reason to use `slugify(self._date_of_birth.strftime("%Y%m%d")` here, or can we instead use just `self._unique_id`? There are probably way more people than me out there and all of their configurations will also break. It is also possible for the users to add the unique_id containing birthdate themselves if any names clash.

Reply:
I'm not really sure why I did this initially, but using `unique_id` directly should work better